### DISCUSSION TITLE
Syslog-ng configuration upgrade

### DISCUSSION THREAD

**Discussion by itheodoridis** [syslog-ng/syslog-ng#4109] [Q&A] [answered]
Hello. I am using syslog-ng in docker format.
Changed to version 3.37 but the config contains a line with : 
`@version: 3.31`
so during init phase it complains that:
`WARNING: Configuration file format is too old, syslog-ng is running in compatibility mode. Please update it to use the syslog-ng 3.37 format at your time of convenience. To upgrade the configuration, please review the warnings about incompatible changes printed by syslog-ng, and once completed change the @version header at the top of the configuration file; config-version='3.31'`
I have searched in order to find what I need to do in order to change the configuration format to current form but I haven't found yet any references to that.
I will keep looking of course but if anyone knows what needs to be done or has a reference, please let me know.

**Comment by MrAnno** [comment, marked as answer]
Once you start v3.37 with a 3.31 config version, it warns you about possible incompatible changes.

Do you see any additional warnings other than the one about the config version?
If not, you can just replace `@version: 3.31` with `@version: 3.37` in your configuration, and the warning will disappear.

If so, those warnings try to explain what needs to be done to remain functionally compatible. They can be confusing sometimes, but you can share those with us and we can help resolve them.

**Reply by itheodoridis** [reply]
No other warnings about the version, just this:
`You have a TLS enabled source without a X.509 keypair. Make sure you have tls(key-file() and cert-file()) options, TLS handshake to this source will fail; location='/etc/syslog-ng/syslog-ng.conf:9:5'`
which makes no sense to me as the only thing in line 9 is the middle line in the following block:
```
source s_network {
    default-network-drivers();
};
```
Is it something about TLS being used by default?
Thank for the answer, I will remove the first line noting the version unless you have additional comments for the other warning.

**Reply by MrAnno** [reply]
Yes, `default-network-drivers()` provides a TLS source as well (basically everything that is needed for legacy and modern (secure) log transports). If you don't need TLS, feel free to ignore the warning.

**Reply by itheodoridis** [reply]
Thank you ! I will follow your advice.
